plugins/vllm/superduper_vllm/model.py

```diff
@@ -1,171 +1,266 @@
-        :param kwargs: The keyword arguments to use.
-        """
-        total_kwargs = {}
-        for key, value in {**self.predict_kwargs, **kwargs}.items():
-            if key in VLLM_INFERENCE_PARAMETERS_LIST:
-                total_kwargs[key] = value
-        return {"prompt": prompt, **total_kwargs}
+from superduper.components.model import Model
 
+if t.TYPE_CHECKING:
+    from vllm import RequestOutput
+    from vllm.entrypoints.chat_utils import ChatCompletionMessageParam
+    from vllm.entrypoints.openai.protocol import ChatCompletionResponse
 
-class _VllmCore:
-    def __init__(self, **kwargs) -> None:
-        # Use kwargs to avoid incompatibility after vllm version upgrade
-        from vllm import LLM
-
-        # Roll back to using the sync engine, otherwise it will no
-        # longer be available on Jupyter notebooks
-        self.engine = LLM(**kwargs)
 
-    def batch_predict(self, prompts: t.List[str], **kwargs):
-        from vllm.sampling_params import SamplingParams
+class _VLLMCore(Model):
+    """Base class for VLLM models.
 
-        sampling_params = SamplingParams(**kwargs)
-        results = self.engine.generate(prompts, sampling_params)
-        n = sampling_params.n
-        texts_outputs = []
-        for result in results:
-            text_outputs = [output.text for output in result.outputs]
-            if n == 1:
-                texts_outputs.append(text_outputs[0])
-            else:
-                texts_outputs.append(text_outputs)
-        return texts_outputs
-
-
-class VllmModel(BaseLLM):
-    """
-    Load a large language model from VLLM.
-
-    :param model_name: The name of the model to use.
-    :param tensor_parallel_size: The number of tensor parallelism.
-    :param trust_remote_code: Whether to trust remote code.
-    :param vllm_kwargs: Additional arguments to pass to the VLLM
-    :param on_ray: Whether to use Ray for parallelism.
-    :param ray_address: The address of the Ray cluster.
-    :param ray_config: The configuration for Ray.
+    :param vllm_params: Parameters for VLLM model
     """
 
-    model_name: str = dc.field(default="")
-    tensor_parallel_size: int = 1
-    trust_remote_code: bool = True
-    vllm_kwargs: dict = dc.field(default_factory=dict)
-    on_ray: bool = False
-    ray_address: t.Optional[str] = None
-    ray_config: dict = dc.field(default_factory=dict)
+    vllm_params: dict = dc.field(default_factory=dict)
 
     def __post_init__(self, db, artifacts):
-        self.on_ray = self.on_ray or bool(self.ray_address)
-        if "tensor_parallel_size" not in self.vllm_kwargs:
-            self.vllm_kwargs["tensor_parallel_size"] = self.tensor_parallel_size
+        super().__post_init__(db, artifacts)
+        assert "model" in self.vllm_params, "model is required in vllm_params"
+        self._async_llm = None
+        self._sync_llm = None
+        tensor_parallel_size = self.vllm_params.get("tensor_parallel_size", 1)
```

Comment:
are we sure these default vars reflect what is in vllm ?

Reply:
This is the default value of vllm package